### ISSUE TITLE
Turning TAA On Then Off Causes Transparency Issues In VR

### ISSUE THREAD

**Issue by ShrikeAlvaron** [Yellow-Dog-Man/Resonite-Issues#3044]
### Describe the bug?

When running Resonite in VR, turning the AA setting to TAA and then off, followed by scaling yourself, causes transparency sorting of materials to break until TAA is turned back on and off again.  This doesn't seem to affect desktop.

### To Reproduce

1. Start Resonite in VR
2. Turn TAA on under graphics settings
3. Immediately switch AA to off (right arrow from the TAA setting)
4. Scale yourself, I find that scaling smaller works best
5. Play around with things that have transparency such as glass materials and UIX) and see how they tend to not work well anymore.
6. Scaling yourself more can adjust the severity of the issues
7. Turn TAA on and back off to fix the issues, until you scale yourself again.

### Expected behavior

Transparency sorting should work properly, or TAA shouldn't even be available in VR in the first place given it's generally broken in VR.

### Screenshots

![Image](https://github.com/user-attachments/assets/aa677677-4bc1-49dd-8553-b5bad0a0fea0)

Note that the inspector in this picture is BEHIND the avatar, yet renders as if it's in front, yet the transparent glasses, eye gloss orbs, and bulb of a devtool are showing up on top of it desperately from the opaque parts of their objects.  The legs are also showing transparency issues with the ground; the entire avatar is raised into the air above the ground.

Also, pictures taken via in-game finger camera came out normal, I had to use a screenshot of the desktop window to show this.

### Resonite Version Number

Beta 2024.10.4.923

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Meta Quest Pro

### Log Files

[BONFIRE - 2024.10.4.923 - 2024-10-05 19_20_28.log](https://github.com/user-attachments/files/17267732/BONFIRE.-.2024.10.4.923.-.2024-10-05.19_20_28.log)

### Additional Context

Readun and Shifty mentioned this issue in #1752 however it doesn't seem a ticket for it was opened then, sorry if I just didn't manage to find it!

### Reporters

Readun (in issue #1752)

**Comment by gameboycjp** [comment]
I've had this happen at random without touching my graphics settings. Glad there's a proper reproduction now!

**Comment by Frooxius** [comment]
Hmm... Does anyone actually actively use TAA in VR?

The quickest solution is to just disallow TAA being used while in VR, since it doesn't seem to play nicely with it. The implementation used is 3rd party and I don't think it'd be worth for us spending time hunting down what causes the bug with it while in VR.

**Comment by TisFoolish** [comment]
No one is likely using it because of the rendering order issues, but when I tried it it was giving better anti-aliasing iirc.

**Comment by shiftyscales** [comment]
I imagine we might have people that would use it if it functioned correctly- but if we aren't going to invest time into fixing it right now, the best we can / should do is disabling it in VR, yeah, @Frooxius.

As-is once it is activated it breaks rendering even after it is switched away from, requiring the user to restart Resonite- so at the least preventing it from being selected in VR should stop the rendering issues it creates. Down the line if we decide we want to explore fixing / replacing it, then we could do so and re-enable that.

**Comment by ModernBalloonie** [comment]
Yeah I feel like this option shouldn't be accessible in VR if it breaks everything.

**Comment by Redd56** [comment]
yeah just had a new user show up and accidnetly trigger this bug, id say it shouldnt be avalible in vr, maybe the content team could make a special edit to the selector in the settings to dissalow it in vr?

**Comment by Redd56** [comment]
> Hmm... Does anyone actually actively use TAA in VR?
> 
> The quickest solution is to just disallow TAA being used while in VR, since it doesn't seem to play nicely with it. The implementation used is 3rd party and I don't think it'd be worth for us spending time hunting down what causes the bug with it while in VR.

okay so i think this is a great option.
i also did a small survey on everyone on my friends list (that replied) + all open sessions and this is my results
25 off
15 fxaa
0 smaa
0 ctaa
0 taa

I really do think that nuking it for vr is the best option. And i hope it gets implimented, because i see multiple new users a week hit taa in vr and then ya gotta tell em "yeah you just gotta reboot the entire game real quick, ya fucked it up by passing by that setting" and it kinda sucks and puts a bad taste in their mouth (atleast it would for me)
